### PR TITLE
Adding L2, Dropout and VIB Regularizer

### PR DESCRIPTION
**Pull request by abhiksingla**
## Why are these changes needed?
Regularization is required to solve the inherent issue of overfitting in the CQL algorithm. The L2 and dropout regularization stand as baselines. Additionally, added support for configuring alpha limits from the trainer config.

- The L2 regularizer will be enabled via trainer config using the field: "enable_l2_regularizer".
- If regularizer is enabled, gamma value can be provided using trainer config and field: "gamma"
- The dropout will be enabled via trainer config using the field "dropout_rate" in "Q_model". If the field is set positive (0.0 < dropout_rate <= 1.0) it will be enabled, otherwise disabled by default.
- Upper and lower limits of lagrangian constraint for alpha value can be set using trainer config using fields: "alpha_upper_bound" and "alpha_lower_bound".

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/cql/loss_functions.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    for weights in model_weights:
+        if not (exclude_bias and "bias" in weights.name):
+            critic_loss += tf.nn.l2_loss(weights)
+
+    if policy_weights is not None:
+        for weights in policy_weights:
+            if not (exclude_bias and "bias" in weights.name):
+                actor_loss += tf.nn.l2_loss(weights)
+
+    return critic_loss + actor_loss
```
Comment (Edilmo) [inline rllib/agents/cql/loss_functions.py]: NIT: end of line missing

File: rllib/agents/sac/sac_tf_policy.py [hunk 2]
```diff
@@ -473,9 +474,18 @@ def make_alpha_actor_losses():
...
+    if l2_regularizer:
+        q_model_weights = policy.model.q_variables()
+        l2_loss = compute_l2_loss(q_model_weights)
+        logger.info(f"Added additional L2 loss")
+        l2_loss = l2_gamma * l2_loss
+
     # In a custom apply op we handle the losses separately, but return them
     # combined in one loss here.
-    return actor_loss + tf.math.add_n(critic_loss) + alpha_loss
+    return actor_loss + tf.math.add_n(critic_loss) + alpha_loss + l2_loss
```
Comment (Edilmo) [inline rllib/agents/sac/sac_tf_policy.py]: I don't see any changes related to VIB, is that intended?

File: rllib/tests/agents/parameters.py [hunk 3]
```diff
@@ -334,6 +334,42 @@ def astuple(self):
             threshold=150.0,
             frameworks=[Framework.TensorFlow],
         ),
+        TestAgentParams.for_moab_move_to_center(
```
Comment (Edilmo) [inline rllib/tests/agents/parameters.py:337]: I don't see neither dropout nor VIB in this test
